models/note.go

```diff
@@ -37,29 +37,132 @@ func (db *DB) StoreNewNote(
 	return NoteId(noteId), nil
 }
 
-//  DB methods
 func (db *DB) GetUsersNotes(userId UserId) (NoteMap, error) {
-	noteMap := make(map[NoteId]*Note)
+	sqlQuery := `
+		SELECT id, author_id, content, creation_time FROM note
+		WHERE author_id = $1`
 
-	{
-		sqlQuery := `
-			SELECT id, author_id, content, creation_time FROM note
-			WHERE author_id = $1`
-		rows, err := db.Query(sqlQuery, int64(userId))
-		if err != nil {
-			return nil, convertPostgresError(err)
+	noteMap, err := db.getNoteMap(sqlQuery, int64(userId))
+	if err != nil {
+		return nil, err
+	}
+
+	return noteMap, nil
+}
+
+func (db *DB) GetAllPublishedNotesVisibleBy(userId UserId) (map[int64]NoteMap, error) {
+
+	sqlQueryIssueNumber := `
+		SELECT COUNT(*) AS IssueNumber FROM publication
+		WHERE publication.author_id = $1`
+
+	var publictionIssueNumber int64
+	if err := db.execOneResult(sqlQueryIssueNumber, &publictionIssueNumber, int64(userId)); err != nil {
+		return nil, err
+	}
+
+	sqlQueryGetNotes := `
+		SELECT
```

Comment:
I spent way more time parsing this sql query than I care to admit. That's largely because i'm bad at sql, but I also have a feeling that this schema is overly complicated. I'm not sure I understand the concept of publishing a note, but if my understanding is correct I think published should just be a boolean column on the notes table rather than its own table.

I'll clarify my understanding of publishing a note over slack before commenting further here.